src/main/java/emu/grasscutter/command/commands/GiveCommand.java

```diff
@@ -212,30 +214,29 @@ public void execute(Player sender, Player targetPlayer, List<String> args) {
     }
 
     private static Avatar makeAvatar(GiveItemParameters param) {
-        return makeAvatar(param.avatarData, param.lvl, Avatar.getMinPromoteLevel(param.lvl), param.constellation);
+        return makeAvatar(param.avatarData, param.lvl, Avatar.getMinPromoteLevel(param.lvl), param.constellation, param.skillLevel);
     }
 
-    private static Avatar makeAvatar(AvatarData avatarData, int level, int promoteLevel, int constellation) {
+    private static Avatar makeAvatar(AvatarData avatarData, int level, int promoteLevel, int constellation, int skillLevel) {
         Avatar avatar = new Avatar(avatarData);
         avatar.setLevel(level);
         avatar.setPromoteLevel(promoteLevel);
+        avatar.changeSkillLevel(skillLevel);
         avatar.forceConstellationLevel(constellation);
-        avatar.recalcStats();
+        avatar.recalcConstellations();
```

Comment:
```suggestion
```
`Avatar::forceConstellationLevel` takes care of this. It also handles `recalcStats`, but it might not call that if it's fed an invalid constellation level so leaving `recalcStats` in is fine.